client/client.go

```diff
@@ -565,6 +565,56 @@ func (c *Client) Do(req *http.Request) (*container.Container, *http.Response, er
 	}
 }
 
+func (c *Client) DoBig(req *http.Request) (*http.Response, error) {
+	// or go home (?)
+	//
+	log.Printf("[DEBUG] Begining DoBig method %s", req.URL.String())
+
+	// retain the request body across multiple attempts
+	for attempts := 0; ; attempts++ {
+		log.Printf("[TRACE] HTTP Request Method and URL: %s %s", req.Method, req.URL.String())
+		resp, err := c.httpClient.Do(req)
+		if err != nil {
+			if ok := c.backoff(attempts); !ok {
+				log.Printf("[ERROR] HTTP Connection error occured: %+v", err)
+				log.Printf("[DEBUG] Exit from Do method")
```

Comment:
Multiple log statement have "Do method", good idea to change the log to match function name